### DISCUSSION TITLE
Keyring-api difficulties

### DISCUSSION THREAD

**Discussion by leopashov** [MetaMask/snaps#1885] [General]
I've been trying to work through the keyring-api example: https://docs.metamask.io/snaps/tutorials/custom-evm-accounts/.
I am getting an error: 'Module '"@metamask/keyring-api"' has no exported member 'buildHandlersChain''.
I am running yarn version 3.6.4 on WSL2, and keyring-api package info is given as: 
@metamask/keyring-api@npm:1.0.0
   ├─ Version: 1.0.0
   │
   └─ Dependencies
      ├─ @metamask/providers@npm:^13.0.0 → npm:13.1.0
      ├─ @metamask/rpc-methods@npm:^3.0.0 → npm:3.0.0
      ├─ @metamask/snaps-utils@npm:^3.0.0 → npm:3.1.0
      ├─ @metamask/utils@npm:^8.1.0 → npm:8.2.0
      ├─ @types/uuid@npm:^9.0.1 → npm:9.0.6
      ├─ superstruct@npm:^1.0.3 → npm:1.0.3
      ├─ uuid@npm:^9.0.0 → npm:9.0.1
      └─ @metamask/snaps-controllers@npm:^3.0.0 → npm:3.1.0 [e2907]
How do I fix this?

I also wanted to ask about the keyring implementation - the documentation (https://github.com/MetaMask/keyring-api/tree/main) says 'Ensure that your keyring implements the [methods called by MetaMask](https://github.com/MetaMask/keyring-api/blob/main/docs/security.md#limit-the-methods-exposed-to-dapps), otherwise some features may not work.', and I see these are defined in the Keyring type, however I cannot find any implementation for these. Do we have to come up with our own logic or is the logic already handled by metamask?

Many thanks for your help!

**Comment by montelaidev** [comment]
Hi @leopashov, the buildHandlersChain was removed in version 0.2.5 and now needs to be implemented on your end.

As for the Keyring type, it's your snap's responsibility to implement these methods. You can refer to this example snap to see how the methods are implemented: https://github.com/MetaMask/snap-simple-keyring/blob/main/packages/snap/src/keyring.ts.

**Reply by leopashov** [reply]
Ahh thank you! Didn't realise that repo existed and was trying to implement keyring functionality by building on the simple snap template
